lms/djangoapps/courseware/masquerade.py

```diff
@@ -46,40 +57,71 @@ def handle_ajax(request, course_key_string):
     role = request_json.get('role', 'student')
     user_partition_id = request_json.get('user_partition_id', None)
     group_id = request_json.get('group_id', None)
+    user_name = request_json.get('user_name', None)
+    if user_name:
+        users_in_course = CourseEnrollment.objects.users_enrolled_in(course_key)
+        try:
+            if '@' in user_name:
+                user_name = users_in_course.get(email=user_name).username
+            else:
+                users_in_course.get(username=user_name)
+        except User.DoesNotExist:
+            return JsonResponse({
+                'success': False,
+                'error': 'There is no user with user name or email address {user_name} '
```

Comment:
"There is no user with ^the^ username or email address <blah> enrolled in this course."